src/container.js

```diff
@@ -49,12 +49,28 @@ export class InvocationHandler {
   */
   invoke(container: Container, dynamicDependencies?: any[]): any {
     return dynamicDependencies !== undefined
-      ? this.invoker.invokeWithDynamicDependencies(container, this.fn, this.dependencies, dynamicDependencies)
-      : this.invoker.invoke(container, this.fn, this.dependencies);
+      ? injectProperties(container, this.fn, this.invoker.invokeWithDynamicDependencies(container, this.fn, this.dependencies, dynamicDependencies))
+      : injectProperties(container, this.fn, this.invoker.invoke(container, this.fn, this.dependencies));
   }
 }
 
 /**
+* Injects property dependencies if the conventional `injectProperties` is defined.
+*/
+function injectProperties(container, fn, instance) {
+  if (fn.injectProperties !== undefined) {
+    let dependencies = fn.injectProperties;
+    for (let property in dependencies) {
+      instance[property] = container.get(dependencies[property]);
+    }
+  }
+  if (instance.postConstruct !== undefined) {
```

Comment:
I'm not sure I like the post construct hook being done this way. Can you elaborate on the purpose of this? It seems like it would only be useful for property injection scenarios, in which case I would expect it to be checked only in the case where there were injected properties.